---
title: "Differential Gene Expression and Pathway Analysis from RNA Sequencing"
author: "Vikas Gubbala"
date: "2/27/2020"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

The purpose of the following excercise is to take processed RNA sequencing data and run differential gene expression analysis (DESeq2) as well as pathway enrichment analysis (from KEGG, GO, Reactome). 

```{r}
#package fo dgea
library (DESeq2)

#packages for annotation
library("AnnotationDbi")
library("org.Hs.eg.db")

#packages for pathway analysis
library(pathview)
library(gage)
library(gageData)
```

## RNA Sequencing Data

In this excercise, we process data from the publication "Trapnell C, Hendrickson DG, Sauvageau M, Goff L et al. "Differential analysis of gene regulation at transcript resolution with RNA-seq". Nat Biotechnol 2013 Jan;31(1):46-53. PMID: 23222703", with GEO number (GSE37704).

RNA sequencing was conducted on control lung fibroblasts and HOX1A knockdown fibroblasts, with three technical replicates for each condition. 

```{r}
metaFile <- "data/GSE37704_metadata.csv"
countFile <- "data/GSE37704_featurecounts.csv"

metadata <- read.csv(metaFile, row.names=1)
head(metadata)

counts <- read.csv(countFile, row.names=1)
counts <- as.matrix(counts[,-1]) #remove the length row
counts <- counts[rowSums(counts) !=0, ] # remove all rows with 0
head(counts)
```
There are `r nrow(counts)` genes in our dataset.

## Differential Gene Expression Analysis

Now, we will run DGEA using the R package DESeq.

```{r}
dds <- DESeqDataSetFromMatrix(countData=counts, colData=metadata,design=~condition)
dds <- DESeq(dds)
res <- results(dds, contrast=c("condition", "hoxa1_kd", "control_sirna"))
res = res[order(res$pvalue),]
```

We will visualize significance and fold change of the differentially expressed genes with a volcano plot.

```{r}
cols <- rep("gray", nrow(res))
cols[abs(res$log2FoldChange)>2] <- "red"
cols[abs(res$log2FoldChange)>2 & res$padj < 0.05] <- "blue"

plot(res$log2FoldChange, -log(res$padj), col = cols, xlab="Log2(FoldChange)", ylab="-Log(P-value)")
```

In our current dataframe we only have Ensemble annotation and gene ID, but KEGG uses Entrez IDs. So we must add this.

```{r}
columns(org.Hs.eg.db)
```

```{r}
#adding the entrez
res$symbol = mapIds(org.Hs.eg.db, keys=row.names(res), keytype="ENSEMBL",column='SYMBOL', multiVals="first")
res$entrez = mapIds(org.Hs.eg.db, keys=row.names(res), keytype="ENSEMBL",column='ENTREZID', multiVals="first")
```

## Pathway Analysis with KEGG

We are going to use the gage package to get enriched pathways, and the pathview package to draw diagrams.

```{r}
data(kegg.sets.hs)
data(sigmet.idx.hs)

#subset to signalling and metabolic pathways
kegg.sets.hs = kegg.sets.hs[sigmet.idx.hs]
#head(kegg.sets.hs, 3)
```

```{r}
#making an array with just the fold changes and entrez as the ID

foldchanges = res$log2FoldChange
names(foldchanges) = res$entrez
head(foldchanges)
```


Now, we will run gage pathway analysis.
```{r}
keggres = gage(foldchanges, gsets = kegg.sets.hs)

#attributes(keggres)

head(keggres$greater)

head(keggres$less)
```

As an example, we will look at the gene set labelled "Cell Cycle."

```{r}
pathview(gene.data=foldchanges, pathway.id="hsa04110")

#PDF Output
#pathview(gene.data=foldchanges, pathway.id="hsa04110", kegg.native=FALSE)

#iterated 
#keggrespathways <- rownames(keggres$greater)[1:5]
#keggresids = substr(keggrespathways, start=1, stop=8)
#pathview(gene.data=foldchanges, pathway.id=keggresids, species="hsa")
```

```{r show = FALSE}
knitr::include_graphics("hsa04110.pathview.png")
```

## Gene Ontology

```{r}
data(go.sets.hs)
data(go.subs.hs)

# subset by biological processes
gobpsets = go.sets.hs[go.subs.hs$BP]
gobpres = gage(foldchanges, gsets=gobpsets, same.dir=TRUE)

head(gobpres$greater)

head(gobpres$less)
```


## Reactome Analysis

Reactome contains information about biological molecules in their relationship to pathways. Here, we will just output a list of signficantly overexpressed genes and use the online Reactome website (https://reactome.org/PathwayBrowser/#TOOL=AT).

```{r}
sig_genes <- res[res$padj <= 0.05 & !is.na(res$padj), "symbol"]
write.table(sig_genes, file="significant_genes.txt", row.names=FALSE, col.names=FALSE, quote=FALSE)
```


## Session Information

```{r}
sessionInfo()
```

